app/model/SubscriptionAcquisitionComponents.scala

```diff
@@ -106,6 +106,15 @@ object SubscriptionAcquisitionComponents {
       printOptions
     }
 
+    def getAbTests(referrerAcquisitionData: Option[ReferrerAcquisitionData]): Option[AbTestInfo] = {
+       val abTest = referrerAcquisitionData.flatMap(_.abTest)
```

Comment:
might be simpler to map `abTest` to `AbTestInfo(Set(ab))` and then do `getOrElse`. and I think you can do this all inside the `.flatMap`

```scala
referrerAcquisitionData.flatMap { r =>
  r.abTest
    .map(ab => AbTestInfo(Set(ab)))
    .getOrElse(r.abTests.map(AbTestInfo(_))
}

```